synapse/events/__init__.py

```diff
@@ -305,49 +293,49 @@ def get_templated_pdu_json(self) -> JsonDict:
 
         return template_json
 
-    def __set__(self, instance, value):
+    def __set__(self, instance, value) -> None:
         raise AttributeError("Unrecognized attribute %s" % (instance,))
 
-    def __getitem__(self, field):
+    def __getitem__(self, field: str) -> Optional[Any]:
```

Comment:
It returns something from `_dict`, which is a `JsonDict`, so it can return `None`, `str`, `int`, `float`, `JsonDict`, or `List` pretty much. We generally just use `Optional[Any]` for these since you can't really express all JSON values using mypy.